moonstone/plot/global_plot.py

```diff
@@ -0,0 +1,265 @@
+        self.df = dataframe
+        self.items_name = items_name
+
+    def plot_mean(self, plotting_options: dict = None, show: Optional[bool] = True, output_file: Optional[str] = False):
+        """
+        method to visualize the mean distribution of the number of reads by items
+
+        :param show: set to False if you don't want to show the plot
+        :param output_file: name of the output file
+        :param plotting_options: options of plotting that will override the default setup \n
+                                 [!] Make sure the value given to an argument is of the right type \n
+                                 options allowed : 'log': `bool` ; 'colorbar': `[str, List[str]]` ;
+                                 'tickangle': `[int, float]`
+        """
+        if plotting_options is None:
+            plotting_options = {}
+        else:
+            plotting_options = _check_types_in_plotting_options(plotting_options)
+
+        plotting_options = _add_x_to_plotting_options(
+            plotting_options, 'log', True)
+        plotting_options = _add_x_to_plotting_options(
+            plotting_options, 'tickangle', -60)
+
+        df_mean = self.df.mean(axis=1)
+        bar_fig = BarGraph(df_mean, plotting_options, show=show, output_file=output_file)
+
+        bar_fig.compute_heterogeneous_bins()
+        bar_fig.in_bins_and_count()    # normalize or not?
+        bar_fig.plot_one_graph(
+            "Distribution of %s mean" % self.items_name,
+            "mean of the number of reads",
+            "number of samples",
+            )
+
+    def _plot_taxonomy_classification(self, level_of_interest, plotting_options: dict = None,
+                                      show: Optional[bool] = True, output_file: Optional[str] = False):
+        """
+        ~~~~ IN CONSTRUCTION (remove _ in title when finished) ~~~~
+        method to visualize the fractions of taxonomic levels (species/genus/...)
+
+        :param level_of_interest: taxonomic classification level (species, genus etc.) to plot
+        :param output_file: name of the output file
+        :param plotting_options: options of plotting that will override the default setup \n
+                                 [!] Make sure the value given to an argument is of the right type \n
+                                 options allowed : 'log': `bool` ; 'colorbar': `[str, List[str]]` ;
+                                 'tickangle': `[int, float]`
+        """
+        if plotting_options is None:
+            plotting_options = {}
+        else:
+            plotting_options = _check_types_in_plotting_options(plotting_options)
+
+        # mean by sample or nb of samples with ?
+
+    # heatmap -> in analysis?
+
+
+class PlotStatsMetadata():
+
+    def __init__(self, metadata_dataframe: pd.DataFrame):
+        self.metadata_df = metadata_dataframe
+
+    def plot_age(self, step=None, plotting_options: dict = None,
```

Comment:
Maybe as discussed, we should probably have only one method here that allow to "plot" one of the column name chosen and perhaps determine in the call whether you want to represent categories, discrete or continue values... I will not create issues at the moment but feel free to create some if you come up with an idea to discuss it :)